ql/src/java/org/apache/hadoop/hive/ql/metadata/DummyPartition.java

```diff
@@ -51,11 +48,15 @@ public DummyPartition(Table tbl, String name) {
     this.name = name;
   }  
 
-  public DummyPartition(Table tbl, String name,
-      Map<String, String> partSpec) {
-    setTable(tbl);
+  public DummyPartition(Table tbl, String name, Map<String, String> partSpec) throws HiveException {
```

Comment:
Can we reuse this object or better create another abstraction, like [Virtual/Hidden]Partition?
cc @kasakrisz

Reply:
we use overriden `getValues` and `getSpec`. And yes, they represent real partitions (not in HMS)

Reply:
`DummyPartition` is used to represent non-HMS partitions. Before, it had just two properties: `table` and `name`. 
But I agree the name is not self explanatory